### PR TITLE
Private Upload Mode - ThreatExchange v2

### PR DESCRIPTION
**Pull request by darkushin**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-4743

## Description
Updating the ThreatExchange v2 integration:
- Added the share_level parameter to the instance configuration and the !url and !domain commands.
- Fixed an issue where the human readable output indicated that no information was retrieved for an indicator, while an exception was raised and the indicator wasn't checked.

### CODE REVIEW

**Review by yuvalbenshalom — APPROVED** [review]

File: Packs/ThreatExchange/ReleaseNotes/2_0_12.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+
+#### Integrations
+
+##### ThreatExchange v2
+
```
Comment (ShacharKidor) [inline Packs/ThreatExchange/ReleaseNotes/2_0_12.md:5]: You also added the *Share Level Type* integration parameter - worth mentioning here.

File: Packs/ThreatExchange/Integrations/ThreatExchangeV2/README.md [hunk 2]
```diff
@@ -284,13 +284,14 @@ Checks URL Reputation
...
 #### Input
 
-| **Argument Name** | **Description** | **Required** |
-| --- | --- | --- |
-| url | URL to be checked. | Required | 
-| limit | The maximum number of results per page. The maximum is 1000. Default is 20. | Optional | 
-| headers | A comma-separated list of headers to display in human-readable format. For example: header1,header2,header3. | Optional | 
-| since | The start timestamp for collecting malware. Supported time formats: epoch time (e.g., 1619870400), ISO 8601 (e.g., 2021-05-01T12:00:00), and free text (e.g., 24 hours ago). | Optional | 
-| until | The end timestamp for collecting malware. Supported time formats: epoch time (e.g., 1619870400), ISO 8601 (e.g., 2021-05-01T12:00:00), and free text (e.g., 24 hours ago). | Optional | 
+| **Argument Name** | **Description**                                                                                                                                                              | **Required** |
```
Comment (ShacharKidor) [inline Packs/ThreatExchange/Integrations/ThreatExchangeV2/README.md:287]: What about adding the new `Share Level Type` parameter to the readme?

File: Packs/ThreatExchange/Integrations/ThreatExchangeV2/ThreatExchangeV2.yml [hunk 3]
```diff
@@ -38,6 +38,17 @@ configuration:
   - F - Reliability cannot be judged
   required: true
   type: 15
+- defaultvalue: 'RED'
```
Comment (ShacharKidor) [inline Packs/ThreatExchange/Integrations/ThreatExchangeV2/ThreatExchangeV2.yml:41]: Shouldn't we get this param in the main func?

**Review by ShacharKidor — CHANGES_REQUESTED** [review]
Nice work!
Added a few comments.

**Review by ShacharKidor — APPROVED** [review]

File: Packs/ThreatExchange/ReleaseNotes/2_0_12.md [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+- Added support for the *share_level* argument in the ***!url*** and ***!domain*** commands.
+- Added support for the *Share Level Type* parameter to the instance configuration.
```
Comment (ShirleyDenkberg) [inline Packs/ThreatExchange/ReleaseNotes/2_0_12.md:7]:
```suggestion
- Added support for the *Share Level Type* parameter in the instance configuration.
```

**Comment by ShirleyDenkberg** [comment]
@yuvalbenshalom @ShacharKidor Doc review completed.

**Pull request merged**